test/resolver.spec.js

```diff
@@ -14,185 +16,129 @@ const fixtureBlockHex = loadFixture('test/fixtures/block.hex')
-        parent: {
-          '/': new CID('z4QJh987XTweMqSLmNakYhbReGF55QBJY1P3WqgSYKsfWuwCvHM')
-        },
-        tx: {
-          '/': new CID('z4QJh987RUUrqpBCCdGuhVMh6Gr21KxokFeHppiNiydiydNEut9')
-        }
-      })
-      done()
-    })
-  })
-
-  it('should return an error if block is invalid', (done) => {
-    IpldZcash.resolver.tree(invalidBlock, (err, value) => {
-      expect(value).to.not.exist()
-      expect(err).to.be.an('error')
-      done()
-    })
-  })
-})
-
-describe('IPLD format resolver API properties', () => {
-  it('should have `multicodec` defined correctly', (done) => {
-    expect(IpldZcash.resolver.multicodec).to.equal('zcash-block')
-    done()
+  it('should return only paths by default', () => {
+    const value = IpldZcash.resolver.tree(fixtureBlockHeader)
+    const paths = [...value]
+    expect(paths).to.have.members([
+      'version',
+      'timestamp',
+      'difficulty',
+      'nonce',
+      'solution',
+      'reserved',
+      'parent',
+      'tx'
+    ])
   })
 
-  it('should have `defaultHashAlg` defined correctly', (done) => {
-    expect(IpldZcash.resolver.defaultHashAlg).to.equal('dbl-sha2-256')
-    done()
+  it('should return an error if block is invalid', () => {
+    expect(() => {
+      IpldZcash.resolver.tree(invalidBlock).next()
+    }).to.throw('Zcash block header needs to be 1487 bytes')
   })
 })
 
-const verifyPath = (block, path, expected, done) => {
-  IpldZcash.resolver.resolve(block, path, (err, value) => {
-    expect(err).to.not.exist()
-    expect(value.remainderPath).is.empty()
-    expect(value.value).to.deep.equal(expected)
-    done()
-  })
+const verifyPath = async (block, path, expected) => {
+  const value = await IpldZcash.resolver.resolve(block, path)
+  expect(value.remainderPath).is.empty()
+  expect(value.value).is.deep.equal(expected)
 }
 
-const verifyError = (block, path, done) => {
-  IpldZcash.resolver.resolve(block, path, (err, value) => {
-    expect(value).to.not.exist()
-    expect(err).to.be.an('error')
-    done()
-  })
+const verifyError = async (block, path, error) => {
+  await expect(
```

Comment:
this can be replaced with a `return` and probably even remove the `async` so it's just a simple Promise returning function

Reply:
I'll keep the `async` as I like to signal that it's returning a Promise. As it's in tests I don't mind the performance implications (in case there are any).